src/Tasks/AssemblyDependency/Resolver.cs

```diff
@@ -181,7 +190,7 @@ ResolutionSearchLocation searchLocation
 
             bool isSimpleAssemblyName = assemblyName?.IsSimpleName == true;
 
-            if (fileExists(pathToCandidateAssembly))
+            if (isFileExistenceVerified || fileExists(pathToCandidateAssembly))
```

Comment:
Can you explain why the `fileExists` callback isn't modified to use the new caching logic based on caching directory listings, and why the `isFileExistenceVerified` flag is being passed around instead?

Reply:
I was trying to isolate changes just to context of 'ResolveFromDirectory' method. 'fileExists' delegate is used by other methods in base and/or derived classes and it would be very hard, I think, to verify that changes in those usages would render better performance and would not cause a regresion. 
Usage of 'isFileExistenceVerified' is also driven by attempt to isolate changes just to `ResolveFromDirectory' method.

Reply:
Gotcha. Still not sure why the check is made a few levels up and not here. It would probably read a little better if instead of `bool isFileExistenceVerified` you passed `bool useDirectoryCache` and this method would call one or the other. To me the way it's written now feels inconsistent.

Reply:
Changed